### PR TITLE
bitcoin-core: Migrate to CMake

### PR DESCRIPTION
**Pull request by hebasto**
This PR migrates to the new CMake-based build system.

### CODE REVIEW

File: projects/bitcoin-core/build.sh [hunk 1]
```diff
@@ -59,21 +59,29 @@ export CPPFLAGS="-D_LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_DEBUG -DBOOST_M
...
 # OSS-Fuzz will provide CC, CXX, etc. So only set:
-# * --enable-fuzz, see https://github.com/bitcoin/bitcoin/blob/master/doc/fuzzing.md
-# * CONFIG_SITE, see https://github.com/bitcoin/bitcoin/blob/master/depends/README.md
+# * -DBUILD_FOR_FUZZING=ON, see https://github.com/bitcoin/bitcoin/blob/master/doc/fuzzing.md
+# * --toolchain, see https://github.com/bitcoin/bitcoin/blob/master/depends/README.md
+EXTRA_BUILD_OPTIONS=
 if [ "$SANITIZER" = "memory" ]; then
   # _FORTIFY_SOURCE is not compatible with MSAN.
-  export CPPFLAGS="${CPPFLAGS} -U_FORTIFY_SOURCE"
+  EXTRA_BUILD_OPTIONS="-DAPPEND_CPPFLAGS='-U_FORTIFY_SOURCE'"
```
Comment (fanquake) [inline projects/bitcoin-core/build.sh:70]: I guess we need to fix this upstream, but one confusing thing here is that this gets added to `APPEND_CPPFLAGS`, but then isn't shown in `CPPFLAGS`, and instead shows up twice in `CXXFLAGS` ?
```bash
Preprocessor defined macros ........... ABORT_ON_FAILED_ASSUME _LIBCPP_HARDENING_MODE=_LIBCPP_HARDENING_MODE_DEBUG BOOST_MULTI_INDEX_ENABLE_SAFE_MODE
C++ compiler flags .................... -O1 -fno-omit-frame-pointer -gline-tables-only -Wno-error=enum-constexpr-conversion -Wno-error=incompatible-function-pointer-types -Wno-error=int-conversion -Wno-error=deprecated-declarations -Wno-error=implicit-function-declaration -Wno-error=implicit-int -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=memory -fsanitize-memory-track-origins -fsanitize=fuzzer-no-link -stdlib=libc++ -std=c++20 -fPIC -Wall -Wextra -Wgnu -Wformat -Wformat-security -Wvla -Wshadow-field -Wthread-safety -Wloop-analysis -Wredundant-decls -Wunused-member-function -Wdate-time -Wconditional-uninitialized -Woverloaded-virtual -Wsuggest-override -Wimplicit-fallthrough -Wunreachable-code -Wdocumentation -Wself-assign -Wundef -Wno-unused-parameter -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -Wstack-protector -fstack-protector-all -fcf-protection=full -fstack-clash-protection -U_FORTIFY_SOURCE
```

Comment (hebasto) [reply projects/bitcoin-core/build.sh:70]: This is a known CMake limitation for `-U...` preprocessor flags. See https://gitlab.kitware.com/cmake/cmake/-/issues/19796.

Comment (fanquake) [reply projects/bitcoin-core/build.sh:70]: Ah. So another years old (unfixed) CMake bug we have to work around.

Comment (fanquake) [reply projects/bitcoin-core/build.sh:70]: Actually, I guess we aren't even working around it, given the incorrect output?

**Review by fanquake — APPROVED** [review]

**Review by DavidKorczynski — APPROVED** [review]

**Comment by Speedtraders48** [comment]
Is the perfect way of life

**Pull request merged**